turbinia/api/models/tasks_statistics.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Turbinia API server Request models."""
+
+import logging
+
+from operator import attrgetter
+from pydantic import BaseModel
+
+from turbinia import client
+from turbinia import state_manager
+from turbinia import config as turbinia_config
+
+log = logging.getLogger('turbinia:api_server:routes:request')
+
+
+class TasksStatistics(BaseModel):
+  """Statistics about classes."""
+  all_tasks: dict = {}
+  successful_tasks: dict = {}
+  failed_tasks: dict = {}
+  requests: dict = {}
+  tasks_per_type: dict = {}
+  tasks_per_worker: dict = {}
+  tasks_per_user: dict = {}
+
+  def get_statistics(
+      self, instance: str, days: int = 0, task_id: str = None,
+      request_id: str = None, user: str = None) -> dict:
+    """Gathers statistics for Turbinia execution data.
+
+      Args:
+        instance (string): The Turbinia instance name (by default the same as the
+            INSTANCE_ID in the config).
+        days (int): The number of days we want history for.
+        task_id (string): The Id of the task.
+        request_id (string): The Id of the request we want tasks for.
+        user (string): The user of the request we want tasks for.
+
+      Returns:
+        task_stats(dict): Mapping of statistic names to values
+      """
+    task_results = state_manager.get_state_manager().get_task_data(
+        instance, days, task_id, request_id, user)
+
+    if not task_results:
+      return {}
+
+    task_stats = {
+        'all_tasks': client.TurbiniaStats('All Tasks'),
+        'successful_tasks': client.TurbiniaStats('Successful Tasks'),
+        'failed_tasks': client.TurbiniaStats('Failed Tasks'),
+        'requests': client.TurbiniaStats('Total Request Time'),
+        # The following are dicts mapping the user/worker/type names to their
+        # respective client.TurbiniaStats() objects.
+        # Total wall-time for all tasks of a given type
+        'tasks_per_type': {},
+        # Total wall-time for all tasks per Worker
+        'tasks_per_worker': {},
+        # Total wall-time for all tasks per User
+        'tasks_per_user': {},
+    }
+
+    # map of request ids to [min time, max time]
+    requests = {}
+
+    for task in task_results:
+      request_id = task.get('request_id')
+      task_type = task.get('name')
+      worker = task.get('worker_name')
+      user = task.get('requester')
+      if not task.get('run_time'):
+        log.debug(
```

Comment:
let's use an f-string here instead of format(), we're trying to do this for the code base in general

Reply:
Replaced here, thanks for noticing that